.github/workflows/spotbugs.yml

```diff
@@ -1,5 +1,12 @@
 name: Spotbugs
-on: [push]
+
+on:
+  push:
+    branches:
+      - master
+  pull_request_target:
+    branches:
+      - master
```

Comment:
how about removing these lines, only running this workflow when there is a push on the master branch?

Reply:
It is necessary that we should inspect the bugs introduce from the pull requests. Otherwise, we need to create more pull requests to fix the problems that unqualified pull requests introduce.

Reply:
will it report the bugs introduced in the PR, block the pr to be merged until these bugs are fixed?

Reply:
No, it won't block right now, it just produces a report of the code. In the future, we could change it to block if there are some violations until we fix the remaining problems.

Reply:
it would be better if there a way to report the issues introduced by the PR individually, which can help the PR author to identify the problems they need to fix.